fe1-web/model/network/JsonRpcResponse.ts

```diff
@@ -1,7 +1,50 @@
-export abstract class JsonRpcResponse {
+import { ProtocolError } from './ProtocolError';
+
+export const UNDEFINED_ID: number = -1;
+
+interface ErrorObject {
+  code: number;
+  description: string;
+}
+
+export class JsonRpcResponse {
+  public readonly result?: number;
+
+  public readonly error?: ErrorObject;
+
   public readonly id: number;
 
   constructor(resp: Partial<JsonRpcResponse>) {
-    this.id = resp.id || -1;
+    this.id = resp.id || UNDEFINED_ID;
+
+    if (resp.error !== undefined && resp.result !== undefined) {
+      throw new ProtocolError('Unexpected json-rpc answer : both \'error\' and \'result\' are present');
+    }
+
+    if (resp.result !== undefined) {
+      if (resp.result) {
+        throw new ProtocolError('Unexpected json-rpc answer : non-zero result value');
+      }
```

Comment:
you've shown me that `[...messages...]` would actually throw an error here :)

Reply:
argh well spotted! I am starting to hate these typescript "unions" ^^'